### PR TITLE
Add blocking suites to daily summary

### PR DESCRIPTION
**Pull request by fejta**
Add a section for blocking suites, and refactor some of the code.

blocking

/cc @rmmh

### CODE REVIEW

File: hack/jenkins/test-history/gen_json.py [hunk 1]
```diff
@@ -136,17 +144,20 @@ def get_tests_from_junit_path(path):
...
     for junit_path in gcs_ls_junit_paths(job, build):
         for test in get_tests_from_junit_path(junit_path):
             yield test
 
-def get_daily_builds(server, prefix):
+
+def get_daily_builds(server, match):
     """Generates all (job, build) pairs for the last day."""
     now = time.time()
+    matcher = re.compile(match)
```
Comment (rmmh) [inline hack/jenkins/test-history/gen_json.py]: put the re.compile in the main function and make get_daily_builds take an arbitrary predicate function?

**Comment by rmmh** [comment]
Overall LGTM

File: hack/jenkins/test-history/gen_json.py (hunk 1 above)
Comment (fejta) [inline hack/jenkins/test-history/gen_json.py]: Done

**Comment by fejta** [comment]
Switched to `blocking = { }`

Can you be more specific about where you think there is too much whitespace? Top-level defs should have two blank lines: https://google.github.io/styleguide/pyguide.html#Blank_Lines

File: hack/jenkins/test-history/gen_html.py [hunk 2]
```diff
@@ -84,129 +90,191 @@ def gen_tests(data, prefix, exact_match):
...
-    return '\n'.join(html), (total_okay, total_unstable, total_failed, total_skipped)
+    return '\n'.join(html), TestMetadata(
+        totals['okay'], totals['unstable'], totals['failed'], totals['skipped'])
+
 
 def html_header():
+    """Return html header items."""
     html = ['<html>', '<head>']
     html.append('<link rel="stylesheet" type="text/css" href="style.css" />')
     html.append('<script src="script.js"></script>')
```
Comment (spxtr) [inline hack/jenkins/test-history/gen_html.py]: I know this isn't introduced by you, but this script shouldn't be in `index.html`. Could you move this line into `gen_html`, probably right before the `</body>` line?

It makes an ugly javascript error that causes the very first bubble to be greyed out.

Comment (fejta) [inline hack/jenkins/test-history/gen_html.py]: Done

**Comment by spxtr** [comment]
LGTM, although I'm a little worried about having tests that aren't run in CI.

**Pull request merged**